### PR TITLE
Google analytics: support release-branch deploys

### PR DESCRIPTION
**Pull request by chalin**
- Contributes to #35299 by enabling GA in `production` and `nonprod` environments
- Followup to #36010

/cc @kbhawkey 
/cc @nate-double-u

### CODE REVIEW

File: layouts/partials/head.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
 {{- template "_internal/opengraph.html" . -}}
 {{- template "_internal/schema.html" . -}}
 {{- template "_internal/twitter_cards.html" . -}}
-{{- if hugo.IsProduction -}}
+{{- if in hugo.Environment "prod" -}}
```
Comment (sftim) [inline layouts/partials/head.html:27]: Can you explain why `hugo.IsProduction` isn't the right test to use? Maybe add a comment here?

Also, perhaps we could logical-OR the two tests and serve the analytics if either is true?

Comment (chalin) [reply layouts/partials/head.html:27]: Thanks for the feedback @sftim. Actually, your comments make me think that we have a broader issue here. Ok, let's hold this while I put together an alternative proposal.

/hold

Comment (chalin) [reply layouts/partials/head.html:27]: I've created the following for further discussion:

- #36333
- #36334

**Comment by kbhawkey** [comment]
👋 @chalin .
Since the PR has been open for over six months, I'll close it.
If you want to continue to work on the changes, you can reopen the pull request. 
/close

**Pull request closed without merging**